var/spack/repos/builtin/packages/exawind/package.py

```diff
@@ -6,35 +6,92 @@
 from spack.package import *
 
 
-class Exawind(CMakePackage):
+class Exawind(CMakePackage, CudaPackage, ROCmPackage):
     """Multi-application driver for Exawind project."""
 
     homepage = "https://github.com/Exawind/exawind-driver"
     git = "https://github.com/Exawind/exawind-driver.git"
 
-    maintainers("jrood-nrel", "psakievich")
+    maintainers("jrood-nrel")
 
     tags = ["ecp", "ecp-apps"]
 
-    version("master", branch="main")
+    # Testing is currently always enabled, but should be optional in the future
+    # to avoid cloning the mesh submodule
+    version("master", branch="main", submodules=True)
+    version("1.0.0", tag="v1.0.0", submodules=True)
 
-    depends_on("trilinos+stk")
-    depends_on("tioga+shared~nodegid")
-    depends_on("nalu-wind+hypre+openfast+tioga+wind-utils")
-    depends_on("amr-wind+hypre+mpi+netcdf+openfast")
-    depends_on("openfast+cxx+shared@2.6.0:")
+    variant("openfast", default=False, description="Enable OpenFAST integration")
+    variant("hypre", default=True, description="Enable hypre solver")
+    variant("stk_simd", default=False, description="Enable SIMD in STK")
+    variant("umpire", default=False, description="Enable Umpire")
+    variant("tiny_profile", default=False, description="Turn on AMR-wind with tiny profile")
+    variant("sycl", default=False, description="Enable SYCL backend for AMR-Wind")
+    variant("gpu-aware-mpi", default=False, description="gpu-aware-mpi")
+
+    conflicts("amr-wind+hypre", when="+sycl")
+
+    for arch in CudaPackage.cuda_arch_values:
+        depends_on("amr-wind+cuda cuda_arch=%s" % arch, when="+cuda cuda_arch=%s" % arch)
+        depends_on("nalu-wind+cuda cuda_arch=%s" % arch, when="+cuda cuda_arch=%s" % arch)
+
+    for arch in ROCmPackage.amdgpu_targets:
+        depends_on("amr-wind+rocm amdgpu_target=%s" % arch, when="+rocm amdgpu_target=%s" % arch)
+        depends_on("nalu-wind+rocm amdgpu_target=%s" % arch, when="+rocm amdgpu_target=%s" % arch)
+
+    depends_on("nalu-wind+tioga")
+    depends_on("amr-wind+netcdf+mpi")
+    depends_on("tioga~nodegid")
     depends_on("yaml-cpp@0.6:")
+    depends_on("nalu-wind+openfast", when="+openfast")
+    depends_on("amr-wind+hypre", when="+hypre~sycl")
+    depends_on("amr-wind~hypre", when="~hypre")
+    depends_on("nalu-wind+hypre", when="+hypre")
+    depends_on("nalu-wind~hypre", when="~hypre")
+    depends_on("amr-wind+sycl", when="+sycl")
+    depends_on("nalu-wind+umpire", when="+umpire")
+    depends_on("amr-wind+umpire", when="+umpire")
+    depends_on("amr-wind+tiny_profile", when="+tiny_profile")
+    depends_on("nalu-wind+gpu-aware-mpi", when="+gpu-aware-mpi")
+    depends_on("amr-wind+gpu-aware-mpi", when="+gpu-aware-mpi")
+    depends_on("nalu-wind@2.0.0:", when="@1.0.0:")
```

Comment:
I'm really torn about these versions. I don't like them so much and debated using a uniform `ecp` tag on github + an `ecp` version.  We are really snapshotting a milestone in the project more than a "release". But I like the idea of composability going forward based off the numbers. Thoughts @jrood-nrel ?